src/detection.c

```diff
@@ -254,3 +257,28 @@ float get_front_sensors_error(void)
 {
 	return distance[SENSOR_FRONT_LEFT_ID] - distance[SENSOR_FRONT_RIGHT_ID];
 }
+
+/**
+ * @brief Detect and return the walls and the spaces around the robot.
+ *
+ * Taking into account that the robot is at least on start position, with the
+ * wheel axis on the beginning of the cell.
```

Comment:
I would omit this. We can actually read the walls around at any point. The function does not actually verify that we are at the correct position, so this is actually handled somewhere else.

Reply:
Ok, I udnerstand that it is the general ideal position of the robot.